packages/visualizations/addon/helpers/default-column-name.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2018, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ */
+import Helper from '@ember/component/helper';
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { metricFormat } from 'navi-data/helpers/metric-format';
+
+export default Helper.extend({
+  bardMetadata: service(),
+
+  compute([column]) {
+    let { type, field } = column
+
+    if (type === 'dateTime') {
+      return 'Date';
+    }
+
+    if (type === 'threshold') {
+      type = 'metric';
```

Comment:
Why do we want to change the `type`? Could we just check to see `if(type === 'metric' || type === 'threshold')`?

Reply:
`metadata.getById ` only supports metric

Reply:
Then we need a different way to do this. I am worried that the column property is getting updated by the helper here, which we do not want.

Reply:
Then we need a different way to do this. I am worried that the column property is getting updated by the helper here, which we do not want.

Reply:
@jkusa @cythrawll @mattgraham1995 What is your option on this?

Reply:
Could you check to see if the helper actually updates the column's type property directly?

Reply:
@mattgraham1995 Yeah. I think we can do that, but I do quite get the convert for changing column property here.

Reply:
I'm just curious if this helper will mutate the type property of a threshold column. 
For example, some threshold column gets passed into this helper and then after the helper returns the default column name, the threshold column's type property has a value of metric.

Reply:
Still waiting on an answer for this.

Reply:
ok.. never mind. I thought destructuring results in a reference. Apparently not. my bad.